server/text_generation_server/models/flash_causal_lm.py

```diff
@@ -820,6 +825,34 @@ def warmup(self, batch: FlashCausalLMBatch):
             self.device,
         )
 
+        if SYSTEM == "rocm":
+            if os.environ.get("PYTORCH_TUNABLEOP_ENABLED", False):
+                if os.environ.get("PYTORCH_TUNABLEOP_TUNING", "1"):
+                    torch.cuda.tunable.tuning_enable(True)
+
+                if os.environ.get("PYTORCH_TUNABLEOP_SEQLENS", False):
+                    tuning_sequences = [int(val) for val in os.environ["PYTORCH_TUNABLEOP_SEQLENS"].split(",")]
+                else:
+                    tuning_sequences = [1, 2, 4, 8, 16, 32]
+                
+                tunableop_filepath = os.path.join("/data", f"tunableop_{self.model_id.replace('/', '-')}_tp{self.world_size}_rank{self.rank}.csv")
+
+                logger.info(f"PyTorch TunableOp (https://github.com/fxmarty/pytorch/tree/2.3-patched/aten/src/ATen/cuda/tunable) is enabled. The warmup may take several minutes, picking the ROCm optimal matrix multiplication kernel for the target lengths {', '.join([str(seqlen) for seqlen in tuning_sequences])} (typical decoding lengths). The picked GEMMs are saved in the file {tunableop_filepath}.")
```

Comment:
How slow is the warmup ? If similar to cuda graphs we should probably make it a default too, no ?

Reply:
We could make it default for seqlens=(1,2,4,8) for example. It may take ~1min